apps/hash-graph/bin/cli/src/subcommand/snapshot.rs

```diff
@@ -0,0 +1,113 @@
+use std::io::Write;
+
+use clap::Parser;
+use error_stack::{IntoReport, Result, ResultExt};
+use graph::{
+    knowledge::Entity,
+    logging::{init_logger, LoggingArgs},
+    store::{
+        crud::Read, query::Filter, test_graph, test_graph::BlockProtocolModuleVersions,
+        DatabaseConnectionInfo, PostgresStorePool, StorePool,
+    },
+};
+use tokio_postgres::NoTls;
+use type_system::{DataType, EntityType, PropertyType};
+
+use crate::error::GraphError;
+
+#[derive(Debug, Parser)]
+pub struct SnapshotDumpArgs;
+
+#[derive(Debug, Parser)]
+pub enum SnapshotCommand {
+    Dump(SnapshotDumpArgs),
+}
+
+#[derive(Debug, Parser)]
+#[clap(version, author, about, long_about = None)]
+pub struct SnapshotArgs {
+    #[command(subcommand)]
+    pub command: SnapshotCommand,
+
+    #[clap(flatten)]
+    pub log_config: LoggingArgs,
+
+    #[clap(flatten)]
+    pub db_info: DatabaseConnectionInfo,
+}
+
+pub async fn snapshot(args: SnapshotArgs) -> Result<(), GraphError> {
+    let _log_guard = init_logger(&args.log_config);
+
+    let pool = PostgresStorePool::new(&args.db_info, NoTls)
+        .await
+        .change_context(GraphError)
+        .map_err(|err| {
+            tracing::error!("{err:?}");
```

Comment:
Shouldn't we just be passing `err` into the tracing macro rather than putting it in a format string?

Reply:
Yes, that would probably be better. I could do this as a drive-by but then I should do this for all subcommands (the code is copied over from there). What do you think?

Reply:
Drive-by seems fine, I'm wondering if there may be an issue here but I'm think the macro falls back to a display/debug implementation. (I'm unclear because of the `Valuable` stuff)

Reply:
Fixed: [Drive-by: Properly print error when acquiring connection](https://github.com/hashintel/hash/pull/2316/commits/e029acc9e1e434febdcdc394010fbf6db6582f8f)